### ISSUE TITLE
Unable to launch debugger in MinGW64 with different terminal path

### ISSUE THREAD

**Issue by dylanweber** [microsoft/vscode-cpptools#10922]
### Environment

- OS and version: Windows_NT x64 10.0.19044
- VS Code: 1.78.0
- C/C++ extension: 1.15.4
- OS and version of remote machine (if applicable): N/A
- GDB / LLDB version: GNU gdb 13.1

### Bug Summary and Steps to Reproduce

Bug Summary: When launching a debug configuration with my default terminal set to `C:\\msys64\\usr\\bin\\bash.exe --login -i` and env var "MSYSTEM=UCRT64;CHERE_INVOKING=1" everything works fine, but when I modify my default terminal path to `C:\\msys64\\msys2_shell.cmd` with args `-no-start -defterm -here -ucrt64` the launch fails due to trying to start with "cmd /C ..." instead of the usual "/usr/bin/env ..."

Steps to reproduce:
1. Have a Windows install with MSYS2 installed in it's default path `C:\msys64`
2. Create a workspace with the following default terminal configuration:
`"MinGW64 UCRT Bash" : {
  "path": "C:\\msys64\\msys2_shell.cmd",
  "args": ["-no-start", "-defterm", "-here", "-ucrt64"]
}`
3. Launch the provided configuration.
4. Observe the terminal launching with `$ cmd /C "c:\Users\...\WindowsDebugLauncher.exe --stdin=.........."` instead of `/usr/bin/env c:\\Users\\...\\WindowsDebugLauncher.exe --stdin=........`
a. Since MinGW converts all paths by default, it will interpret "/C" to mean "C:\\" and thus the actual command being ran is `cmd C:\ "c:\Users\...\WindowsDebugLauncher.exe -stdin=........"` which only results in a new command prompt being opened, not in the string argument getting executed.
5. Observe the error "Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information."

### Debugger Configurations

```shell
{
	"name": "(gdb) Launch tests",
	"type": "cppdbg",
	"request": "launch",
	"program": "${workspaceFolder}\\build\\tester.exe",
	"args": [],
	"stopAtEntry": true,
	"cwd": "${workspaceFolder}\\build",
	"environment": [
		{
			"name": "PATH",
			"value": "C:\\msys64\\ucrt64\\bin\\;${env:PATH}"
		}
	],
	"externalConsole": false,
	"MIMode": "gdb",
	"miDebuggerPath": "C:\\msys64\\ucrt64\\bin\\gdb.exe",
	"setupCommands": [
		{
			"description": "Enable pretty-printing for gdb",
			"text": "-enable-pretty-printing",
			"ignoreFailures": true
		}
	],
	"logging": {
		"engineLogging": false
	}
}
```

### Debugger Logs

```shell
--> E (output): {"type":"event","event":"output","body":{"category":"stderr","output":"\r\n\r\n"},"seq":8}
 
 
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (20456) Send Event AD7MessageEvent\r\n"},"seq":10}
1: (20456) Send Event AD7MessageEvent
--> E (output): {"type":"event","event":"output","body":{"category":"telemetry","output":"VS/Diagnostics/Debugger/Launch","data":{"VS.Diagnostics.Debugger.ImplementationName":"Microsoft.MIDebugEngine","VS.Diagnostics.Debugger.EngineVersion":"17.4.30208.2","VS.Diagnostics.Debugger.HostVersion":"17.4.30208.2","VS.Diagnostics.Debugger.AdapterId":"cppdbg","VS.Diagnostics.Debugger.Launch.ErrorCode":1005,"VS.Diagnostics.Debugger.Launch.IsError":true}},"seq":12}
--> R (launch-2): {"type":"response","request_seq":2,"success":false,"command":"launch","message":"Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information.","body":{"error":{"id":1005,"format":"Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information."}},"seq":14}
```

### Other Extensions

CMake Tools v1.14.31

**Comment by WardenGnaw** [comment]
Thank you for reporting this issue. We’ll let you know if we need more information to investigate it.

**Comment by adrien-lenglet** [comment]
I have a fix on the user end for that issue. I found that the path to the terminal needs to include the word `bash` in order for the invocation of `WindowsDebugLauncher.exe` to assume a Bash shell.

So on my end, I created the following file at `C:\msys64\usr\bin\bash_mingw64.bat` to keep it short:
```bat
@ECHO OFF

set MSYSTEM=MINGW64
set CHERE_INVOKING=1

C:\msys64\usr\bin\bash.exe --login -i
```
Then, setting the default terminal to `C:\msys64\usr\bin\bash_mingw64.bat` did the trick. This should work for other `MSYSTEM`s.

I was unsuccessful at locating the root cause of that behavior.
